### PR TITLE
Safe restock

### PR DESCRIPTION
**Pull request by dvir001**
## About the PR
Added new restock type, secured.
When attempted to open by force it will destroy the refill content.

Right now its assigned to 3 new refill crates.
Astro - 1000
Circuit - 3000
Ammo - 4000

This is all still subject to change and update, the prices, labels and crate types are not final.

**Changelog**
:cl:
- add: New type of refill, secured.
- add: 3 new crates.

### CODE REVIEW

**Comment by dvir001** [comment]
If its not working ill probably have to update VendingMachineRestockTest.cs for a bigger change.

**Comment by Cheackraze** [comment]
didnt get a chance to test yet but a quick glance looked good so far. see if i get some time tomorrow and do an in-game test just to confirm.

**Comment by Cheackraze** [comment]
lgtm but theres some merge conflicts now, just gotta resolve and submit for review we can get it in

**Comment by dvir001** [comment]
Done, its working as intended for now.

**Comment by Cheackraze** [comment]
since we are doing it, we really should just replace them all and make them all worth only 100 because they are all kind of just broken

**Comment by dvir001** [comment]
> since we are doing it, we really should just replace them all and make them all worth only 100 because they are all kind of just broken

Here

File: Resources/Locale/en-US/prototypes/catalog/cargo/cargo-vending.ftl [hunk 1]
```diff
@@ -31,9 +31,15 @@ ent-CrateVendingMachineRestockRobustSoftdrinks = { ent-CrateVendingMachineRestoc
 ent-CrateVendingMachineRestockSalvageEquipment = { ent-CrateVendingMachineRestockSalvageEquipmentFilled }
     .desc = { ent-CrateVendingMachineRestockSalvageEquipmentFilled.desc }
 
+ent-CrateVendingMachineRestockAstroVend = { ent-CrateVendingMachineRestockAstroVendFilled }
```
Comment (Cheackraze) [inline Resources/Locale/en-US/prototypes/catalog/cargo/cargo-vending.ftl:34]: lets keep all of our localization stuff separate, it makes it easier for translation and also avoids merge conflicts when we update.

File: Resources/Prototypes/Catalog/Cargo/cargo_vending.yml [hunk 2]
```diff
@@ -126,6 +126,16 @@
   category: Engineering
   group: market
 
+- type: cargoProduct
```
Comment (Cheackraze) [inline Resources/Prototypes/Catalog/Cargo/cargo_vending.yml:129]: Same here with our own custom product entries, lets keep them separate in a new file in the _nf folder

File: Resources/Prototypes/Entities/Objects/Specific/Service/vending_machine_restock.yml [hunk 3]
```diff
@@ -46,10 +49,59 @@
         acts: [ "Destruction" ]
 
 - type: entity
-  parent: BaseVendingMachineRestock
+  parent: BaseItem
```
Comment (Cheackraze) [inline Resources/Prototypes/Entities/Objects/Specific/Service/vending_machine_restock.yml:52]: reparenting all of the existing ones to ours is fine, and adding staticprices to cut on exploits is as well, but similar to the other yml prototypes. Lets put our custom ones in separate files in _nf so it becomes easier to merge/maintain and find our custom items. The modifications to the existing ones in the file are ok

**Review by Cheackraze — APPROVED** [review]
everything looks fine, but lets keep our custom yml and ftl stuff separated in the file structure so its easier to maintain/translate. Will be good to get this in and we can lower the vending stocks a bit on some of them safely

**Pull request merged**